test/strings.jl

```diff
@@ -1280,20 +1280,77 @@ end
 @test isxdigit("a") == true
 @test isxdigit("g") == false
 
-@test is_valid_ascii("is_valid_ascii") == true
-@test is_valid_ascii("Σ_not_valid_ascii") == false
-@test is_valid_char('a') == true
-@test is_valid_char('\x00') == true
-@test is_valid_char('\ud800') == false
-
-@test is_valid_utf16(utf16("a")) == true
-@test is_valid_utf16(utf16("\ud800")) == false
-# TODO is_valid_utf8
-
 # Issue #11140
-@test is_valid_utf32(utf32("a")) == true
-@test is_valid_utf32(utf32("\x00")) == true
-@test is_valid_utf32(UInt32[0xd800,0]) == false
+@test isvalid(utf32("a")) == true
+@test isvalid(utf32("\x00")) == true
+@test isvalid(UTF32String, UInt32[0xd800,0]) == false
+
+# Issue #11241
+
+@test isvalid(ASCIIString, "is_valid_ascii") == true
+@test isvalid(ASCIIString, "Σ_not_valid_ascii") == false
+@test isvalid(Char, 'a') == true
+@test isvalid(Char, '\x00') == true
+
+# test all edge conditions
+@test isvalid(Char, 0) == true
```

Comment:
It would be cleaner to have this as a loop, no?
```
for (i, v) in ((0,true), (0xd7ff,true), (0xd800,false), ........)
    @test isvalid(Char, i) == v
end
```
and similarly for the other tests?

My general rule of thumb is that whenever you find yourself copying-and-pasting code, you are making a mistake.